yt_dlp/extractor/youtube.py

```diff
@@ -4552,19 +4552,43 @@ def process_language(container, base_url, lang_code, sub_name, query):
             'uploader_id': channel_handle,
             'uploader_url': format_field(channel_handle, None, 'https://www.youtube.com/%s', default=None),
         })
+
+        def get_pacific_tz():
+            # Python 3.8 should be deprecated soon
+            # This fallback may make the timestamp slightly inaccurate for 3.8 users.
+            if sys.version_info < (3, 9):
+                return dt.timedelta(hours=-7)
+            from zoneinfo import ZoneInfo
+            return dt.datetime.now(ZoneInfo('US/Pacific')).utcoffset()
+
+        # We only want timestamp IF it has second precision
+        # Additionally, if there is no timezone present, we should assume it is in PT.
+        timestamp = (
+            parse_iso8601(get_first(microformats, 'uploadDate'), timezone=get_pacific_tz())
+            or parse_iso8601(search_meta('uploadDate'), timezone=get_pacific_tz())
+        )
+        upload_date = (
+            dt.datetime.fromtimestamp(timestamp, dt.timezone.utc).strftime('%Y%m%d') if timestamp else
+            (
+                unified_strdate(get_first(microformats, 'uploadDate'))
+                or unified_strdate(search_meta('uploadDate'))
+            ))
```

Comment:
If `timestamp`, don't set `upload_date` at all. Core code will do it

Reply:
We still have the old upload_date fallback below. Additionally it helps us avoid having to rewrite all the below code